### DISCUSSION TITLE
unique species and language switching

### DISCUSSION THREAD

**Discussion by snaterendesnader** [mcguirepr89/BirdNET-Pi#379] [Ideas]
It seems that the determination of unique species is language dependent, and doesn't use the latin species name as the primary key. This causes statistics to get skewed when you change the system language (in my case from English to Dutch).

So, all statistics now display a mix of Dutch and English species names, which are counted as different species. This could be remedied by using the latin name to group species and use the localized name for displaying purposes.

**Comment by Clumpton** [comment]
I would like to +1 this as I have discovered this behaviour when switching to French from English. It would appear that the database index is LatinName_CommonNameInCurrentLanguage - so indeed you will get new entries for the same species if you change the database language.
I put forward that this is not at all scientific and that the common name should come dynamically from a language lookup table.
If, at a later date, we wanted to make a multilingual interface this would speed things up too.
On the lighter side I would like to point out that Human_Human does not follow the naming schema, surely it should be Homo Sapiens_Human?

**Comment by ehpersonal38** [comment]
Unfortunately there's just so much that depends on the Com_Name field in the database. There are over 130 entries of code alone that involve it, and our file structure system means that all bird sound extractions are saved in `~/BirdSongs/.../Com_Name/`, making this task even harder to achieve. But it's certainly doable so if anyone has the time to convert everything over (abolish Com_Name in the DB entirely, use Sci_Name for everything and have a Com_Name lookup from ~/BirdNET-Pi/scripts/labels.txt on the fly for UI stuff), I will definitely welcome pull requests.

**Reply by Clumpton** [reply]
OK, I see the problem. If I broke something and become housebound for a few weeks I could take a shot at it, but for the time being her indoors is moving into full gardening mode and my time is very limited.
